airbyte-integrations/connectors/source-freshdesk/source_freshdesk/api.py

```diff
@@ -138,15 +138,57 @@ def _api_get(self, url: str, params: Mapping = None):
     def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
         """Iterate over entities"""
 
+    @property
+    def name(self):
+        """Name of the stream"""
+        stream_name = self.__class__.__name__
+        if stream_name.endswith("API"):
+            stream_name = stream_name[:-3]
+        return stream_name
+
     def read(self, getter: Callable, params: Mapping[str, Any] = None) -> Iterator:
         """Read using getter"""
         params = params or {}
+        stream = self.name
+
+        # This block extends TicketsAPI Stream to overcome '300 page' server error.
+        # Since the TicketsAPI Stream list has a 300 page pagination limit, after 300 pages, update the parameters with
+        # query using 'updated_since' = last_record, if there is more data remaining.
+        if stream == "Tickets":
```

Comment:
this should be an override of the `read` method on the tickets stream rather than a conditional in the super class. 

Also, can you verify if other streams will have the same issue? if so, we might want to update the super class method to handle it for all cases

Reply:
Tickets - is the only stream that has this limitation, since it has the biggest amount of data. 
Ok, will try to override the read method.

Reply:
Done